core-java/src/main/java/org/spine3/server/aggregate/Aggregate.java

```diff
@@ -274,18 +297,56 @@ public void restore(SnapshotOrBuilder snapshot) {
         setState(stateToRestore, snapshot.getVersion(), snapshot.getWhenModified());
     }
 
+    private void putUncommitted(EventRecord record) {
+        uncommittedEvents.add(record);
+    }
+
     /**
-     * @return immutable view of records for uncommitted events
+     * Returns all uncommitted events (including state-neutral).
+     *
+     * @return immutable view of records for all uncommitted events
+     * @see #getStateNeutralEventClasses()
      */
     @CheckReturnValue
-    public List<EventRecord> getUncommittedEvents() {
+    public List<EventRecord> getAllUncommittedEvents() {
         return ImmutableList.copyOf(uncommittedEvents);
     }
 
     /**
-     * Returns and clears the events that were uncommitted before the call of this method.
+     * Returns uncommitted events (excluding state-neutral).
+     *
+     * @return immutable view of records for applicable uncommitted events
+     * @see #getStateNeutralEventClasses()
+     */
+    public Collection<EventRecord> getApplicableUncommittedEvents() {
+        //noinspection LocalVariableNamingConvention
+        final Set<Class<? extends Message>> stateNeutralEventClasses = getStateNeutralEventClasses();
+        final Predicate<EventRecord> isNotStateNeutral = isNotStateNeutralPredicate(stateNeutralEventClasses);
+        final Collection<EventRecord> result = filter(uncommittedEvents, isNotStateNeutral);
+        return result;
+    }
+
+    @SuppressWarnings("MethodParameterNamingConvention") // to be precise
+    private static Predicate<EventRecord> isNotStateNeutralPredicate(
```

Comment:
How about `isStateChanging`?

Have a constant to avoid creation of predicate instances all the time. And do have the method that returns the constant.